internal/transformer/classes/generators/dockerfilegenerators/PHPdockerfilegenerator.go

```diff
@@ -0,0 +1,207 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dockerfilegenerators
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	gopache "github.com/Akash-Nayak/GopacheConfig"
+	"github.com/konveyor/move2kube/environment"
+	"github.com/konveyor/move2kube/internal/common"
+	"github.com/konveyor/move2kube/types/qaengine/commonqa"
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	phpExt      = ".php"
+	virtualHost = "VirtualHost"
+)
+
+// PHPDockerfileGenerator implements the Transformer interface
+type PHPDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// PhpTemplateConfig implements Php config interface
+type PhpTemplateConfig struct {
+	Ports    []int32
+	ConfFile string
+}
+
+// Init Initializes the transformer
+func (t *PHPDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *PHPDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *PHPDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// DetectConf extracts the port information from .conf file
+func DetectConf(dir string) (int32, string, error) {
+	var port int32
+	var confFileName string
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return port, confFileName, err
+	}
+	for _, de := range dirEntries {
+		ext := filepath.Ext(de.Name())
+		if ext != ".conf" {
+			continue
+		}
+		confFileName := de.Name()
+		confFilePath := filepath.Join(dir, de.Name())
+		confFile, err := os.Open(confFilePath)
+		if err != nil {
+			logrus.Errorf("Could not open the conf file: %s", err)
+			return port, confFileName, err
+		}
+		root, err := gopache.Parse(confFile)
+		if err != nil {
+			logrus.Errorf("Error while parsing configuration file : %s", err)
+			return port, confFileName, err
+		}
+		match, err := root.FindOne(virtualHost)
+		if err != nil {
+			logrus.Errorf("Could not find the VirtualHost in conf file: %s", err)
```

Comment:
Is virtual host mandatory in a conf file? If not, it should not be a error

Reply:
I think it's not mandatory, so I will change it.